Remove max-width constraint on areas container

zmarties opened the pull request:
With this in place, the areas grid only shows 3 items per row, which is wasting space on wider screens.
The equivalent css for the integrations tab does not have the max-width style, and thus efficiently uses the full width.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->

Fix the layout of the areas dashboard grid so that it uses the full width of the screen.

<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->

Simple change to remove an extraneous style from the CSS.

<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Checklist

<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

zmarties committed:
Remove max-width constraint on areas container

With this in place, the areas grid only shows 3 items per row, which is wasting space on wider screens.
The equivalent css for the integrations tab does not have the max-width style, and thus efficiently uses the full width.

bramkragten commented:
It's done this way on purpose, because it's easier for the eye to look vertically than horizontally. @matthiasdebaat maybe you can have a look?

zmarties commented:
Unfortunately the eye can only scan the items that are on the screen.  With a limit of 3 items per row, my screen shows 3 rows, thus 9 areas.  Removing the width limit allows the display of 3 rows each of 5 items, thus 15 areas in total (which is sufficient to show all of my areas) - it's clearly much better to show all the areas without scrolling than it is be forced to scroll to see the ones I want. (If I change my browser to show less chrome I can get 4 rows in, which thus shows 20 areas).

Two questions
- do we have analytics that show how many areas most people have - is it more than the 9 that would fit on a standard HD sized screen if the restricted width is kept?
- the areas are shown in alphabetical order, with no chance to rearrange them.  Thus the least interesting areas typically come first - Bedrooms and bathrooms come before the more important lounge, movie room, kitchen, or office!  Are there any plans to allow the areas to be arranged in a different order?

bramkragten commented:
I think we can at least bump the max width to around 1200px or something, allowing 4 items per row. But would love some guidelines from our UX designer on this.

matthiasdebaat commented:
Hi @zmarties, can you share a screenshot of what you mean?

I don't have a strong opinion about the number of items that can be displayed. Our premise is that entities are summarized, such as your lamps, switches, etc. How many entity types are there?

If we look at it from a responsive design perspective, the title and items should not overlap. On larger screens, that probably wouldn't happen, but on a smartphone, we need to look at that. A solution might be to make the last option `mdi:dots-vertical` with an overflow menu of items that don't fit.

bramkragten commented:
@matthiasdebaat I think you misunderstood, it is about the width of the page/container containing area card, it is now limited to 1000px width, @zmarties suggests to remove that and have the container be the full width of the page.

matthiasdebaat commented:
> @matthiasdebaat I think you misunderstood, it is about the width of the page/container containing area card, it is now limited to 1000px width, @zmarties suggests to remove that and have the container be the full width of the page.

Ooh, thanks! Let's use the Lovelace sizing 4 * 500 max. We should avoid [these kind of stretching](https://github.com/home-assistant/frontend/pull/10544#issuecomment-962852474).

zsarnett commented:
@zmarties I think @matthiasdebaat is suggesting 2000px. Can you make this change?

zmarties committed:
Update ha-config-areas-dashboard.ts

Alternative solution asked for by UX designer - rather than always using the full width, limit it to the numerical value that is 4 times the max-width of the individual containers.
